### PR TITLE
Support attributes and excludeAttributes query params for List roles v2 endpoint

### PR DESCRIPTION
**Pull request by ChanikaRuchini**
**Purpose**
Support attributes and excludeAttributes query params for List roles v2 endpoint

**Related Issues**
https://github.com/wso2/product-is/issues/18096

### CODE REVIEW

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerV2.java [hunk 1]
```diff
@@ -593,19 +599,81 @@ private String getSearchFilter(String attributeName, String filterOperation, Str
...
+                }
+                if (requiredAttributes.contains(ASSOCIATED_APPLICATIONS)) {
+                    // Set associated applications.
+                    List<MultiValuedComplexType> associatedApps =
+                            convertAssociatedAppsToMultivaluedComplexType(role.getAssociatedApplications());
+                    if (CollectionUtils.isNotEmpty(associatedApps)) {
+                        scimRole.setAssociatedApplications(associatedApps);
+                    }
+                }
+            }
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerV2.java:676]: let's extract this to separate private method to improve readability

File: components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/RoleResourceV2.java [hunk 2]
```diff
@@ -162,7 +164,8 @@ public Response getRoles(@HeaderParam(SCIMProviderConstants.ACCEPT_HEADER) Strin
             // Create charon-SCIM role endpoint and hand-over the request.
             RoleResourceV2Manager roleResourceManager = new RoleResourceV2Manager();
             SCIMResponse scimResponse = roleResourceManager
-                    .listWithGETRole(roleManager, filter, startIndex, count, sortBy, sortOrder);
+                    .listWithGETRole(roleManager, filter, startIndex, count, sortBy, sortOrder, attribute,
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/RoleResourceV2.java:167]: shouldn't a similar change be done for getRolesByPOST ?

File: scim2.yaml [hunk 3]
```diff
@@ -1353,6 +1353,16 @@ paths:
           description: The order in which the "sortBy" parameter is applied.
           required: false
           type: string
+        - name: attributes
+          in: query
+          description: SCIM defined attributes parameter.
```
Comment (dewniMW) [inline scim2.yaml:1358]: let's avoid saying it as SCIM defined as the endpoint is not defined in the spec

File: components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/RoleResourceV2.java (hunk 2 above)
Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/RoleResourceV2.java:167]: No, in the 'getRolesByPOST' method, attributes and excludeAttributes are sent in the request body, not as request parameters.

File: scim2.yaml (hunk 3 above)
Comment (ChanikaRuchini) [reply scim2.yaml]: Fixed

**Review by dewniMW — APPROVED** [review]

**Review by dewniMW — APPROVED** [review]

**Pull request merged**